egs/wsj/s5/utils/lang/make_subword_lexicon_fst.py

```diff
@@ -0,0 +1,240 @@
+#!/usr/bin/env python3
+
+# 2019 Dongji Gao
+
+from collections import namedtuple
+from make_lexicon_fst import read_lexiconp
+import argparse
+import math
+import sys
+
+# see get_args() below for usage mesage
+def get_args():
+    parser = argparse.ArgumentParser(description="""This script creates a 
+        position depent subword lexicon FST from a origin lexicon""")
+    parser.add_argument('--sil-phone', type=str, help="""Text form of
+        optional-silence phone.""")
+    parser.add_argument('--sil-prob', type=float, default=0.0, help="""Probability
+        of silence between words (including the beginning and end of word sequence).
+        Must be in range [0.0, 1.0]""")
+    parser.add_argument('--sil-disambig', type=str, help="""Disambiguation symbol
+        to disambiguate silence, e.g. #5.""")
+    parser.add_argument('--position-dependent', action="store_true", help="""Whether the input lexicon
+        is position-dependent.""")
+    parser.add_argument("--separator", type=str, default="@@", help="""Separator indicates the
+        position of a subword in word.""")
+    parser.add_argument('lexiconp', type=str, help="""Filename of lexicon with
+        pronunciation probabilities (normally lexiconp.txt), with lines of the
+        form 'word prob p1 p2...', e.g. 'a, 1.0 ay'""")
+    args = parser.parse_args()
+    return args
+
+def contain_disambig_symbol(phones):
+    """Disambig symbol is at the end of phones in the form of #1, #2..."""
+    return True if phones[-1][0] == "#" else False
+
+def print_arc(src, dest, phone, word, cost):
+    print('{}\t{}\t{}\t{}\t{}'.format(src, dest, phone, word, cost))
+
+def is_end(word, separator):
+    return (not word.endswith(separator))
+
+def get_suffix(phone):
+    return phone[-2:]
+
+def get_position(phones, constant_variable):
```

Comment:
There is no documentation here:  What does this function return?  What is the meaning of the possible values it can return?